src/View/Helper/GoogleMapHelper.php

```diff
@@ -847,7 +847,12 @@ public function icon($url, array $options = []) {
 		// The shadow image is larger in the horizontal dimension
 		// while the position and offset are the same as for the main image.
 		if (empty($options['size'])) {
-			$data = getimagesize($url);
+			if (substr($url, 0, 1) === '/') {
```

Comment:
This will probably not work for URLS like ``//...` ( https://www.paulirish.com/2010/the-protocol-relative-url/ ).

Reply:
(oops, putting this back on the thread)
hmmm. it seems like the protocol relative url is intended to avoid using http if the site was accessed as https, without having to specify the scheme again for the asset in the url. but regardless of whether the site is accessed by http or https, wouldn't the file under the webroot still be available on the file system? the addition of webroot was because the asset doesn't have a http or https in front (which caused getimagesize to fail), so wouldn't the getimagesize still work okay on the local file for the double slash case too?

Reply:
You can try, but the above code then woudnt be good.

Reply:
Hey, sorry I lost track of this issue for a bit.
I don't see a problem with the implementation yet.  If there's a double slash in front of the $url, then the resulting file system path (with WWW_ROOT in front) will also have a double slash in it.  But that should be just fine in every file system I know of, Unix or Windows.  All the slashes should, in effect, collapse to a single one for local file system access.  Is there another issue I'm not seeing?

Reply:
It sounds a bit unclean, that is the only concern then I guess.

Reply:
I hear that concern, and I have modified the code to call "realpath" on the file system path before passing it to the getimagesize function.
Hopefully this gets it looking reasonable.  I have verified that the most recent check-in works properly on my site.
Are there any other issues I should address?
Thanks.  -- Chris

Reply:
can realpath return false or sth?

Reply:
yep, that could happen.  i've updated the code again with error checking for that case and a fallback to the simple url.